tests/Bridge/Symfony/Routing/CachedRouteNameResolverTest.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Tests\Bridge\Symfony\Routing;
+
+use ApiPlatform\Core\Bridge\Symfony\Routing\CachedRouteNameResolver;
+use ApiPlatform\Core\Bridge\Symfony\Routing\RouteNameResolverInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Prophecy\Argument;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * @author Teoh Han Hui <teohhanhui@gmail.com>
+ */
+class CachedRouteNameResolverTest extends \PHPUnit_Framework_TestCase
+{
+    public function testConstruct()
+    {
+        $cacheItemPoolProphecy = $this->prophesize(CacheItemPoolInterface::class);
+
+        $decoratedProphecy = $this->prophesize(RouteNameResolverInterface::class);
+
+        $cachedRouteNameResolver = new CachedRouteNameResolver($cacheItemPoolProphecy->reveal(), $decoratedProphecy->reveal());
+
+        $this->assertInstanceOf(RouteNameResolverInterface::class, $cachedRouteNameResolver);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage No item route associated with the type "AppBundle\Entity\User".
+     */
+    public function testGetRouteNameForItemRouteWithNoMatchingRoute()
+    {
+        $cacheItemProphecy = $this->prophesize(CacheItemInterface::class);
+        $cacheItemProphecy->isHit()->willReturn(false)->shouldBeCalled();
+
+        $cacheItemPoolProphecy = $this->prophesize(CacheItemPoolInterface::class);
+        $cacheItemPoolProphecy->getItem(Argument::type('string'))->willReturn($cacheItemProphecy);
+        $cacheItemPoolProphecy->save($cacheItemProphecy)->shouldNotBeCalled();
+
+        $decoratedProphecy = $this->prophesize(RouteNameResolverInterface::class);
+        $decoratedProphecy->getRouteName('AppBundle\Entity\User', false)
```

Comment:
Not for a fictional class... :laughing:

Perhaps I should change this to `Some\Test\Dummy`?